src/core/internal/vararg/sysv_x64.d

```diff
@@ -0,0 +1,304 @@
+                    ap.stack_args += T2.sizeof.alignUp;
+                }
+            }
+            else static if (is(T1 == double) || is(T1 == float))
+            {
+                void* a = void;
+                if (ap.offset_fpregs < (6 * 8 + 16 * 8) &&
+                    ap.offset_regs < 6 * 8 && T2.sizeof <= 8)
+                {
+                    *p1 = *cast(T1*) (ap.reg_args + ap.offset_fpregs);
+                    ap.offset_fpregs += 16;
+                    a = ap.reg_args + ap.offset_regs;
+                    ap.offset_regs += 8;
+                }
+                else
+                {
+                    *p1 = *cast(T1*) ap.stack_args;
+                    ap.stack_args += T1.sizeof.alignUp;
+                    a = ap.stack_args;
+                    ap.stack_args += 8;
+                }
+                // Be careful not to go past the size of the actual argument
+                const sz2 = T.sizeof - 8;
+                (cast(void*) p2)[0..sz2] = a[0..sz2];
+            }
+            else static if (is(T2 == double) || is(T2 == float))
+            {
+                if (ap.offset_regs < 6 * 8 && T1.sizeof <= 8 &&
+                    ap.offset_fpregs < (6 * 8 + 16 * 8))
+                {
+                    *p1 = *cast(T1*) (ap.reg_args + ap.offset_regs);
+                    ap.offset_regs += 8;
+                    *p2 = *cast(T2*) (ap.reg_args + ap.offset_fpregs);
+                    ap.offset_fpregs += 16;
+                }
+                else
+                {
+                    *p1 = *cast(T1*) ap.stack_args;
+                    ap.stack_args += 8;
+                    *p2 = *cast(T2*) ap.stack_args;
+                    ap.stack_args += T2.sizeof.alignUp;
+                }
+            }
+            else // both in regular registers
+            {
+                void* a = void;
+                if (ap.offset_regs < 5 * 8 && T1.sizeof <= 8 && T2.sizeof <= 8)
+                {
+                    *p1 = *cast(T1*) (ap.reg_args + ap.offset_regs);
+                    ap.offset_regs += 8;
+                    a = ap.reg_args + ap.offset_regs;
+                    ap.offset_regs += 8;
+                }
+                else
+                {
+                    *p1 = *cast(T1*) ap.stack_args;
+                    ap.stack_args += 8;
+                    a = ap.stack_args;
+                    ap.stack_args += 8;
+                }
+                // Be careful not to go past the size of the actual argument
+                const sz2 = T.sizeof - 8;
+                (cast(void*) p2)[0..sz2] = a[0..sz2];
+            }
+
+            return result;
+        }
+        else
+        {
+            static assert(false);
+        }
+    }
+    else
+    {
+        static assert(false, "not a valid argument type for va_arg");
+    }
+}
+
+///
+void va_arg()(va_list ap, TypeInfo ti, void* parmn)
+{
+    TypeInfo arg1, arg2;
+    if (!ti.argTypes(arg1, arg2))
```

Comment:
Please invert the condition check and move the assert here s.t. the function has less indentation.